how_to_add_new_hotspot.md

```diff
@@ -0,0 +1,142 @@
+
+Hotspots needed to be accessed via Bluetooth sometimes. Especially at the beginning, onboarding time. But keeping the required Bluetooth interface on all the time imposes two problems. First, it is a security thread and leaks some information to trace the device. Second it also effects power consumption negatively as the device is expected to work 7/24.
+
+This input triggers the Bluetooth advertisement process and then the user can access it via Bluetooth interface.
+
+#### ECCOB
+Indicates whether the ECC is onboard or not. It is a bool parameter.
+
+#### TYPE
+Defines Helium Miner type (e.g. Full, Lite etc.)
+
+#### CELLULAR
+Defines whether the device has cellular (mobile) hardware or not. It's a bool parameter.
+
+#### FCC_IDS, IC_IDS
+ID's given to the device by the regulatory bodies.
+
+#### CONTAINS_FCC_IDS, CONTAINS_IC_IDS
+The regulatory body ID's which could be found on inner modules like a cellular modem or concentrator module.
+
+## Adding automation steps
+If the new hotspot type would need a new fleet, as we do right now, it would need some changes on our automation pipeline.
+
+The developer should add necessary steps into the GitHub Action files which could be found [here](https://github.com/NebraLtd/helium-miner-software/tree/master/.github/workflows)
+
+## Modifying settings.ini
+The new hotspot could need to define another I2C bus path. Then it needed to be added into the settings.ini file and `gen_docker_compose.py` file needs to be updated, if necessary.
+
+## Defining configuration parameters in the Balena Fleet
+A single board computer generally wouldn't have a BIOS due to constraints. Therefore manufacturers like Raspberry Pi Foundation created a clever way to keep boot parameters on the device. For example there is a `config.txt` file under `/boot` for Rasspberry Pi SBC's and it defines the parameters which is cruical to make the device work properly.
+
+In Balena, such critical parameters are defined in Fleet Configuration and passed down to the device. So before adding a new device type into a new fleet configuration parameters need to be set in Fleet Configuration.
+
+## Defining variant and frequency in the Belana Fleet
+There are two critical parameters which has to be defined in a Balena Fleet before adding a new device into it.
+
+### VARIANT
+This parameter defines which variant will be used in the whole firmware stack. The name is the key from the `variant_definitions` dictionary which is detailed above.
+
+**Example:**
+| Name | Value | Service 
+|------|-------|--------
+| VARIANT | NEBHNT-IN1 | All Services
+
+### FREQ
+Like in many things, the world also has been divided about the RF frequency spectrum. So fleets are separated also with their center frequencies because hotspots acts differently regarding to their RF bands.
+
+If this parameter wouldn't be defined in a fleet, the `packet_forwarder` container could not progress further and waits for this parameters in a restart loop.
```

Comment:
this is not strictly true.

the FREQ actually does nothing.

REGION_OVERRIDE can override the region, however we only use this on testnet fleets for testing (but even then, it is technically unnecessary if the miner has been asserted on the blockchain - as if it has been asserted it just gets the region from the blockhain)

Reply:
While working with RAK, without this parameter it was stalled an kept in a restart loop. Maybe I should go back and check the behavior again. Let's keep it and change if strictly necessary to remove.

Reply:
I guess we should take a look at this parameter in another task. (#385 maybe?)